packages/babel-plugin-set-component-name/src/index.js

```diff
@@ -0,0 +1,28 @@
+module.exports = ({ template, types: t }) => {
+  return {
+    visitor: {
+      Program(path) {
+        path.get('body').forEach(nodePath => {
+          const { node } = nodePath;
+          if (
+            // componentName.propTypes = ...
+            t.isExpressionStatement(node) &&
+            t.isAssignmentExpression(node.expression, { operator: '=' }) &&
+            t.isMemberExpression(node.expression.left)
+          ) {
+            // componentName.propTypes
+            const expr = node.expression.left;
+            if (
+              t.isIdentifier(expr.object) &&
+              t.isIdentifier(expr.property, { name: 'propTypes' })
+            ) {
+              nodePath.insertBefore(
+                template.ast(`${expr.object.name}.displayName = 'Mui${expr.object.name}';`),
```

Comment:
In https://github.com/mui-org/material-ui/pull/15023 the name is prefixed with `Mui`, is that wanted here as well?